unified-scheduler-pool/src/lib.rs

```diff
@@ -751,11 +770,33 @@ impl TaskHandler for DefaultTaskHandler {
 
         let pre_commit_callback = match scheduling_context.mode() {
             BlockVerification => None,
-            BlockProduction => Some(|processing_result: &'_ Result<_>| {
-                if let Err(error) = processing_result {
-                    Err(error.clone())?;
+            BlockProduction => Some(|processing_result: &'_ Result<ProcessedTransaction>| {
+                let Ok(processed_transaction) = processing_result else {
+                    return Err(processing_result.as_ref().unwrap_err().clone());
                 };
 
+                // Now it's the procrastinated time of _optimistic_ provisioning of block cost at the
```

Comment:
Have wanted to rework our other pipeline to do this as well - reserve-commit scheme is bad imo: https://github.com/anza-xyz/agave/issues/3528